docs/linters.md

```diff
@@ -0,0 +1,210 @@
+# Linters
```

Comment:
Maybe up here I would add a short paragraph on linter philosophy:

* Speed; we want it to be nice to run it in a pre-commit hook, so we run linters in parallel and support incremental checks
* Accuracy: We try hard to catch as many common mistakes as possible either via a linter or (when a linter can't do it) some sort of end-to-end a test case.  
* Completeness: Our goal is to have most common style issues by caught by the linters, so new contributors to the codebase can efficiently fix produce code with correct style without needing to go back-and-forth with a reviewer.